### PR TITLE
Fix improperly tokenized testcases causing minimization result to not reproduce error.

### PR DESCRIPTION
**Pull request by mpherman2**
This should stop any round of minimization where the tokenizer has failed. It logs when the tokenizer fails so that we can continue to amend the grammar as we see common missed tokens.

### CODE REVIEW

**Comment by mpherman2** [comment]
This is complicated to discuss here, so I will explain more in person. If I move the checks to minimize task the checks will be significantly more complicated, and errors will have worse consequences. HTML minimizer in particular just sends its data to other minimizers so it would be very hard to check the tokenizer until you are in the minimizer code.

**Comment by mpherman2** [comment]
Done with refactoring. PTAL

File: src/python/bot/minimizer/chunk_minimizer.py [hunk 1]
```diff
@@ -32,6 +33,8 @@ def __init__(self, *args, **kwargs):
   def _execute(self, data):
     """Minimize |data| using the algorithm from CF (but backwards)."""
     testcase = minimizer.Testcase(data, self)
+    if testcase.get_current_testcase_data() != data:
+      raise errors.TokenizationFailureError("JSMinimizer")
```
Comment (Dor1s) [inline src/python/bot/minimizer/chunk_minimizer.py]: Is `JSMinimizer` a correct string here? Same for the delta minimizer class.

Also, please use single quotes for string literals

**Pull request merged**